mongodb/docs-worker-pool
# [DOP-3911]: Check project path for monorepo

## Ticket
[DOP-3911](https://jira.mongodb.org/browse/DOP-3911)

## Description

Adds logic to find the project's path within the monorepo based on given file changes. This change supports detecting nested projects as well. 

This PR also includes the functionality to check whether or not the project paths detected from GitHub match what is configured within the corresponding `docset` entry.

This PR also includes small bugfixes/improvements:
- fixed bug when updating the webhooks for a feature branch
- added the creating of a parameter store parameter within the CDK for the feature flag

## Testing 

1. Create a fork of the [docs-monorepo](https://github.com/10gen/docs-monorepo) repository
2. follow the instructions for configuring the webhook [here](https://wiki.corp.mongodb.com/pages/viewpage.action?spaceKey=DE&title=How-To%3A+Use+Snooty%27s+Autobuilder+to+Build+Your+Content) with the following modifications: Use the following webhook URL: `https://3kxa3rkxf8.execute-api.us-east-2.amazonaws.com/prod/webhook/githubEndpoint/trigger/build` and the secret for pre-prod GitHub push jobs found on Parameter Store.
3. Make a pull request with an arbitrary change on the forked repo under the `docs-landing` directory
4. Verify that the correct repo path is found by checking the [CloudWatch log](https://us-east-2.console.aws.amazon.com/cloudwatch/home?region=us-east-2#logsV2:log-groups/log-group/$252Faws$252Flambda$252Fauto-builder-stack-enhanc-apigithubTriggerLambdaA2-tXwcwvrXd5dA) for the lambda
![image](https://github.com/mongodb/docs-worker-pool/assets/20285369/1e341c74-210d-467e-a964-5782fcafd4d4)


NOTE: You will receive an error message stating that the build has failed. This is due to the autobuilder not having permission to clone the `docs-monorepo` repository.
